core/src/main/java/org/mule/runtime/core/api/context/notification/FlowStackElement.java

```diff
@@ -40,13 +42,27 @@ public String getProcessorPath() {
   public String getFlowName() {
     return flowName;
   }
+  
+  /**
+   * @return the time when the flow stack was created.
+   */
+  public Long getCreationTime() {
+    return creationTime;
+  }
+  
+  /**
+   * @return the milliseconds elapsed between its creation and now.
+   */
+  public Long getElapsedTime() {
+    return System.currentTimeMillis() - creationTime;
+  }
 
   @Override
   public String toString() {
     if (processorPath == null) {
-      return format("%s", flowName);
+      return format("%s %d ms", flowName, getElapsedTime());
```

Comment:
Is this only for the dump scenario? I'm concerned that in the error logs where this is shown the log will be cluttered with numbers. Do you have examples of how would it look? both for the dump and the error log

Reply:
That will be added for dumps and also for error messages when you enable verbose exceptions.
This is an example from a dump:
```
"eventId": "0-58af4b70-d7ac-11e8-9cf1-88e9fe8670c4151711677913640924701204632292736282969839189247101584149156809491559108404346167228189123471093442353",
"flowStack": [
    "complexFlow1(complexFlow1/processors/0/route/0/processors/0 @ complex:complex.xml:29 (Execute)) 1914 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1916 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1919 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1921 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1923 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1925 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1927 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1930 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1932 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1933 ms",
    "complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 1935 ms",
    "complexFlow(complexFlow/processors/1 @ complex:complex.xml:24 (Flow Reference)) 1938 ms"
]
```
Here is an example from an error log:
```
ERROR 2018-10-24 17:43:20,743 [[MuleRuntime].cpuLight.11: [complex].complexFlow1.CPU_LITE @393786b1] [event: 0-e7700da0-d7ab-11e8-9cf1-88e9fe8670c4] org.mule.runtime.core.internal.exception.OnErrorPropagateHandler: 
********************************************************************************
Message               : java.lang.RuntimeException: fail.
Element               : complexFlow1/processors/0/route/0/processors/0 @ complex:complex.xml:29 (Execute)
Element XML           : <scripting:execute engine="groovy" doc:name="Execute" doc:id="0c957ea2-9d93-4321-89c0-3e7e357e7ec0">
                        <scripting:code>Thread.sleep(5000);
                        throw new RuntimeException('fail');</scripting:code>
                        </scripting:execute>
Error type            : SCRIPTING:EXECUTION
FlowStack             : at complexFlow1(complexFlow1/processors/0/route/0/processors/0 @ complex:complex.xml:29 (Execute)) 5010 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5011 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5013 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5015 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5017 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5020 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5023 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5026 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5028 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5029 ms
                        at complexFlow1(complexFlow1/processors/0/route/1/processors/1 @ complex:complex.xml:45 (Flow Reference)) 5031 ms
                        at complexFlow(complexFlow/processors/1 @ complex:complex.xml:24 (Flow Reference)) 5035 ms
Payload               : null
Payload Type          : null
--------------------------------------------------------------------------------
```

To me, it doesn't add a lot of numbers and it adds some valuable information. I added full log examples in https://www.mulesoft.org/jira/browse/MULE-15701.